docs/specification.md

```diff
@@ -370,30 +373,112 @@ when verification was in progress locally. If this is the case, the provisional
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
+
+Bob does not use gittuf and pushes to a branch. Alice notices this as her gittuf
+client detects a push to the branch without an accompanying RSL entry. She
+validates that the change came from Bob and creates an RSL entry on his behalf,
+identifying him and including information about how she verified it was him.
+Alice applies M2.
+
+#### An incorrect RSL entry is added
+
+If an entry is malformed (structurally), Git should catch it if it's not a valid
+commit. In addition, gittuf must catch any valid commits with malformed RSL
+data. For example, a necessary field may be missing, or may contain invalid
+data. gittuf must reject such entries and warn the user when they are
+encountered. As this scenario requires the author's client to be buggy or
+malicious, it cannot be trusted to catch the invalid entries when the actor
+pushes to the main copy of the repository. However, correct gittuf clients
+verify that RSL entries are not invalid when they pull from the main copy of
+the repository. If the main copy is enabled with gittuf, such incidents can be
+caught before other users receive the invalid changes.
+
+If the main copy is not enabled with gittuf, when a user detects such an
+incident, they can use the mechanisms described in M1 to invalidate the
+incorrect entry.
+
+Bob has a buggy gittuf client and pushes an invalid entry to the main copy of
+the repository. Alice pulls and receives a warning from her gittuf client. Alice
+reverses Bob's changes, creating an RSL entry for the affected branch if needed,
+and includes an RSL annotation skipping Bob's RSL entry.
+
+#### A gittuf access control policy is violated
+
+Bob creates an RSL entry for a branch he's not authorized for by gittuf policy.
+He pushes a change to that branch.  Alice notices this and (TODO: decide if
+alice needs to be authorized) she's authorized by policy for the affected
+branch. Alice reverses Bob's change, creating a new RSL entry reflecting that.
+Alice also creates an RSL annotation marking Bob's entry as one to be skipped.
+Alice, therefore, uses M1. Finally, if the policy violation was because of a key
+compromise, the policy is updated to revoke and rotate the affected key.
+
+#### Attacker modifies or deletes historical RSL entry
+
+Overwriting or deleting an historical RSL entry is a complicated proposition.
+Git's content addressable properties mean that a SHA-1 collision is necessary to
+overwrite an existing RSL entry in the Git object store. Further, the attacker
+also needs more than push access to the repository as Git will not accept an
+object it already has in its store (TODO: double check). Similarly, deleting an
+entry from the object store preserves the RSL structure cosmetically but
+verification workflows that require the entry will fail.
```

Comment:
Okay, but what is the impact here?  You've talked about why it is hard to pull off, but what happens if it is done?

Reply:
Does it read better now?

Reply:
I understand, but don't feel like you are convincing about the actual impact this has.  So, if verification workfloes that require the entry fail, what do you do?  Is this okay?

Reply:
Indicated that detection is followed by restoring the affected repository from other, local copies of the repository.